docs/design/navigation-and-scale.md

```diff
@@ -0,0 +1,143 @@
+# UX and navigation that scales
+
+This design document is a proposal for Dashboard UI user experience that
+covers navigation and the way of displaying and interacting with K8s objects.
+
+## Background
+
+Currently Dashboard UI allows users to see and interact with a very limited
+subset of Kubernetes resources. For example, it shows Replication Controllers,
+while ignoring existence of Replica Sets, Deployments, Daemon Sets, etc.
+Additionally, it does not allow for creating or modifying individual
+resources, while supporting only application creation (akin to `kubectl run`)
+and scaling. For more details see [mockups](mockups/11-11-2015-initial) of
+current implementation.
+
+These limitations have been number one source of feature requests and
+user complaints (e.g., #564, #524, #509, #510, #232). The core team also
+considers solving these limitations as a natural evolution of the UI.
+
+## Problem statement
+
+* Scale the UX to support displaying and editing all current and future
+  Kubernetes resources (e.g., replica sets, daemon sets, container, nodes,
+  volumes, secrets, deployments, etc.).
+* Scale for the number of objects displayed (currently: cards are suitable for
+  O(20), desired: show any number of objects).
+* Guide users through use-case based views (don't just do API dump).
+* Make the UI aware of namespaces and multi-cluster installations.
+
+# Design
+The design is based on a few novelties to the UX of Kubernetes Dashboard. This
+is:
+* introduction of use-case based navigation menu and views
+* replacement of free-form cards by tabularized resource lists
+* being explicit about Kubernetes resources displayed in the UI
+* introduction of action bar for performing all actions on resources
+
+## Navigation
+Navigation is realized by left hand side menu. The menu consists of use case
+driven categories that group Kubernetes resources. For example, there may be
+category called "Apps" which deals with Replication Controllers, Pods,
+Deployments, etc., and "Config" which deals with Secrets, Volumes,
+Namespaces, etc. The exact definition of categories will happen in a separate
+design.
+
+![Navigation](mockups/23-03-2016-scale-and-navigation/navigation.png)
+
+Menu categories may be expanded into raw resource subpages for expert users.
+
+![Menu extended](mockups/23-03-2016-scale-and-navigation/menu-extended.png)
+
+Menu follows standard responsiveness practises where it folds into icons and
+hamburger menu when space is limited.
+
+![Mobile navigation](mockups/23-03-2016-scale-and-navigation/mobile-navigation.png)
+
+## View templates
+
+This section shows concepts used as templates for all views.
+
+View which does not deal with a resource or a resource list, e.g., landing
+page which shows an overview of a cluster. The page consists of, potentially
+dismissable, blocks that display data.
```

Comment:
recommendation, keep all help links in one single location / helper utility in the code and reference them based on unique topic names everywhere else, because inevitably doc links will change and its much easier to update a link in one place, see https://github.com/openshift/origin/blob/master/assets/app/scripts/filters/util.js#L168  and  https://github.com/openshift/origin/blob/master/assets/app/scripts/constants.js